File: bounded-contexts/java/src/main/java/p1/myshop/shipping/services/DeliveryService.java
package p1.myshop.shipping.services;

import p1.myshop.catalog.entities.HasWeight;
import p1.myshop.catalog.services.CatalogService;
import p1.myshop.shoppingcart.entities.Cart;
import p1.myshop.shoppingcart.entities.CartItem;

public class DeliveryService {
    private final CatalogService catalogService;

    public DeliveryService(CatalogService catalogService) {
        this.catalogService = catalogService;
    }

    public double calculateOrderWeight(Cart cart) {
        return cart.items().stream()
                .map(CartItem::itemId)
                .map(catalogService::loadItem)
                .mapToDouble(HasWeight::weight)
                .sum();
    }
}

File: bounded-contexts/dotnet/MyShop.Services/Shipping/DeliveryService.cs
using MyShop.Services.ShoppingCart;
using System;
using System.Linq;
using System.Collections.Generic;
using System.Text;

namespace MyShop.Services.Shipping
{
    public class DeliveryService
    {
        public DeliveryService(ICatalogService catalogService)
        {
            CatalogService = catalogService;
        }

        public ICatalogService CatalogService { get; }

        public double CalculateOrderWeight(Cart cart)
        {
            return cart.GetItems()
                .Sum(item => item.Quantity * CatalogService.LoadItem(item.ItemId).Weight);
        }
    }
}
